linux57-tkg/install.sh

```diff
@@ -132,74 +167,118 @@ if [ "$1" = "install" ]; then
+    export CCACHE_NOHASHDIR="true"
+    msg2 'ccache was found and will be used'
+
   fi
 
-  _kernel_flavor="${_kernel_localversion}"
   if [ -z $_kernel_localversion ]; then
     _kernel_flavor="tkg-${_cpusched}"
+  else
+    _kernel_flavor="tkg-${_kernel_localversion}"
   fi
 
-  if [ "$_distro" = "Ubuntu" ]; then
+  if [ "$_distro" = "Ubuntu" ]  || [ "$_distro" = "Debian" ]; then
+
     if make -j ${_thread_num} deb-pkg LOCALVERSION=-${_kernel_flavor}; then
       msg2 "Building successfully finished!"
+
+      cd "$_where"
+
+      # Create rpms folder if it doensn't exist
+      if ! [ -d DEBS ]; then
+        mkdir DEBS
+      fi
+      
+      # Move rpm files to RPMS folder inside the linux-tkg folder
+      mv "$_where"/*.deb "$_where"/DEBS/
+
       read -p "Do you want to install the new Kernel ? y/[n]: " _install
-      if [[ $_install =~ [yY] ]] || [[ $_install =~ [yY] ]] || [ $_install = "yes" ] || [ $_install = "Yes" ]; then
+      if [[ $_install =~ [yY] ]] || [ $_install = "yes" ] || [ $_install = "Yes" ]; then
         cd "$_where"
         _kernelname=$_basekernel.$_kernel_subver-$_kernel_flavor
-        _headers_deb=linux-headers-${_kernelname}*.deb
-        _image_deb=linux-image-${_kernelname}_*.deb
+        _headers_deb="linux-headers-${_kernelname}*.deb"
+        _image_deb="linux-image-${_kernelname}_*.deb"
+        _kernel_devel_deb="linux-libc-dev_${_kernelname}*.rpm"
         
-        sudo dpkg -i $_headers_deb $_image_deb
+        sudo dpkg -i DEBS/$_headers_deb DEBS/$_image_deb DEBS/$_kernel_devel_deb
+      fi
+    fi
 
-        # Add to the list of installed kernels, used for uninstall
-        if ! { [ -f installed-kernels ] && grep -Fxq "$_kernelname" installed-kernels; }; then
-          echo $_kernelname >> installed-kernels 
-        fi   
+  elif [[ "$_distro" = "Fedora" ||  "$_distro" = "Suse" ]]; then
+
+    # Replace dashes with underscores, it seems that it's being done by binrpm-pkg
+    # Se we can actually refer properly to the rpm files.
+    _kernel_flavor=${_kernel_flavor//-/_}
+
+    if make -j ${_thread_num} rpm-pkg EXTRAVERSION="_${_kernel_flavor}"; then
+      msg2 "Building successfully finished!"
+
+      cd "$_where"
+
+      # Create rpms folder if it doensn't exist
+      if ! [ -d RPMS ]; then
+        mkdir RPMS
+      fi
+      
+      # Move rpm files to RPMS folder inside the linux-tkg folder
+      mv ~/rpmbuild/RPMS/x86_64/* "$_where"/RPMS/
+
+      #Clean up the original folder, unneeded and takes a lot of space
+      rm -rf ~/rpmbuild/
+
+      read -p "Do you want to install the new Kernel ? y/[n]: " _install
+      if [ "$_install" = "y" ] || [ "$_install" = "Y" ] || [ "$_install" = "yes" ] || [ "$_install" = "Yes" ]; then
+        
+        _kernelname=$_basekernel.${_kernel_subver}_$_kernel_flavor
+        _headers_rpm="kernel-headers-${_kernelname}*.rpm"
+        _kernel_rpm="kernel-${_kernelname}*.rpm"
+        _kernel_devel_rpm="kernel-devel-${_kernelname}*.rpm"
```

Comment:
Would it be possible to get the name of the RPMs without using globbing? I'm concerned about globbing being able to capture more than one filename.

Reply:
The compilation adds for example `+-1.x86_64` at the end of the package names that is different if RPM or DEB, it shouldn't blob more that one file since the `$_kernelname` variable contains the kernel version + `tkg-pds` or `tkg-${customname}` if the user gave one. Though I think it is possible to write the filename exactly.

Reply:
Ah, that's rather annoying. I suppose this is fine as is, as matching the exact filename sounds like a pain, and, thinking about it more, if it were to glob more than one file somehow, it wouldn't be the end of the world.

Reply:
Yea I think it is okay, and the script uses the distro's package manager, that shows what's about to be installed and asks for confirmation. We'll see if someone opens an issue about that, I will figure something out at that time :P